### ISSUE TITLE
Migration help to 1.0.0 regarding conanfile_directory

### ISSUE THREAD

**Issue by SirLemyDanger** [conan-io/conan#2302]
Hi,

 we are in the process of migrating to conan 1.0.0 from 0.27.
One of the things which changed was the removal of self.conanfile_directory and the requirement to call conan install & build with the path to the conanfile.

My setup:
qmake
the qmake .pro file sits in the same folder as the conanfile.py
```
bin/
src/
projects/
projects/conanfile.py
projects/makefile.pro
```

In the makefile.pro I need to know where the generator output from conan is to include it and I need to know where the makefile.pro is to give the rigth cwd to this cmd:
`self.run("qmake %s" % (self.buildfile_name), cwd=???)`

I have problems to locate the two without self.conanfile_directory. Espeacially beacuase the new conan build/install syntax encoutage people to call those cmds from different locations.
To be more clear: 
I cannot find a generic solution if somebody calls conan install/build from projects/ or from one folder up projects/..

Cheers

**Comment by memsharded** [comment]
Hi @SirLemyDanger 

If ``self.conanfile_directory`` was being used in the ``build()`` method, most likely you can replace it with ``self.source_folder``. 

Regarding the call to ``conan install``, in most cases it can be replaced with ``conan install .``, same for ``conan build``

Is this what you are asking? I would need some more details about the exact usage of it. If you have private details you don't want to share, please write us an email, or use the # conan slack channel. Thanks!

**Comment by SirLemyDanger** [comment]
Yes thanks. Could you maybe clarify a bit more (in the docs) what the new self.xxx_folders do? The documentation is quite understandable what those variables are/default to in the local cache but outside of the local cache its was not clear to me.

Have I understood it correct that the `self.source_folder` defaults to the location where the conanfile is, and `self.build_folder` is the folder in which I called `conan install`

**Comment by memsharded** [comment]
Sure, I am adding an item in this docs issue: https://github.com/conan-io/docs/issues/458. Please follow up there for docs improvement.

Could this issue be closed now? Did you manage to upgrade to 1.0 and completely get rid of ``conanfile_directory``? Thanks for your feedback.

**Comment by SirLemyDanger** [comment]
Yes thanks. The issue can be closed
